File: test/fixtures/ruby/post_with_colon_in_header.rb
require 'net/http'

uri = URI('http://localhost:28139/endpoint')
req = Net::HTTP::Post.new(uri)
req.content_type = 'application/json'
req['key'] = 'abcdefg'

req_options = {
  use_ssl: uri.scheme == "https",
}
res = Net::HTTP.start(uri.hostname, uri.port, req_options) do |http|
  http.request(req)
end

File: test/fixtures/kotlin/post_with_colon_in_header.kt
import java.io.IOException
import okhttp3.OkHttpClient
import okhttp3.Request
import okhttp3.RequestBody.Companion.toRequestBody

val client = OkHttpClient()

val request = Request.Builder()
  .url("http://localhost:28139/endpoint")
  .method("POST", "".toRequestBody())
  .header("Content-Type", "application/json")
  .header("key", "abcdefg")
  .build()

client.newCall(request).execute().use { response ->
  if (!response.isSuccessful) throw IOException("Unexpected code $response")
  response.body!!.string()
}
